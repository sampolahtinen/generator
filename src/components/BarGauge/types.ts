export interface BarGaugeProps {
  value: number;
}
