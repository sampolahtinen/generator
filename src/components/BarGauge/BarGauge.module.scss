@import '../../styles/colors.scss';

.percentageGauge {
  position: relative;
  background: $grey_100;
  height: 8px;
  width: 100%;
  border-radius: 16px;

  span {
    border-radius: 16px;
    position: absolute;
    background-color: $grey_300;
    top: 0;
    left: 0;
    height: inherit;
    width: 0;
    opacity: 1;
    transition: width 300ms linear, opacity 300ms ease 300ms;
  }
}
