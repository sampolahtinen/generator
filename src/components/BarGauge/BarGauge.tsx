import React, { useRef, useEffect, useState } from 'react';
import styles from './BarGauge.module.scss';
import { BarGaugeProps } from './types';

const BarGauge = ({ value }: BarGaugeProps) => {
  const [pixelValue, setPixelValue] = useState(0);
  const gaugeRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    const gaugeElement = gaugeRef.current;
    if (gaugeElement) {
      const gaugeWidth = gaugeElement?.offsetWidth;
      const pixels = (value / 100) * gaugeWidth;
      setPixelValue(pixels);
    }
  }, [value]);

  return (
    <div className={styles.percentageGauge} ref={gaugeRef}>
      <span style={{ width: pixelValue }} />
    </div>
  );
};

export default BarGauge;
