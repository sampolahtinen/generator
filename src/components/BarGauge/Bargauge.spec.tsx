import React from 'react';
import { mount } from 'enzyme';
import BarGauge from './BarGauge';

describe('BarGauge Component', () => {
  const wrapper = mount(
    <div style={{ width: '100px' }}>
      <BarGauge value={50} />
    </div>
  );

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
