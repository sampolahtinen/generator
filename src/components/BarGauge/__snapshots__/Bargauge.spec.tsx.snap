// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarGauge Component renders 1`] = `
<div
  style={
    Object {
      "width": "100px",
    }
  }
>
  <BarGauge
    value={50}
  >
    <div
      className="percentageGauge"
    >
      <span
        style={
          Object {
            "width": 0,
          }
        }
      />
    </div>
  </BarGauge>
</div>
`;
